index.bs

```diff
@@ -371,28 +405,173 @@ model]]:
+    1. Set |navigable|'s [=top-level traversable/bounce tracking record=] to a new
+        [=bounce tracking record=] with [=bounce tracking record/initial host=]
+        set to |initialHost|.
+1. Otherwise,
+    1. If |sourceSnapshotParams|'s <a spec="html">has transient activation</a> is true:
+        1. Run [=record stateful bounces for bounce tracking=] given |navigable|'s [=navigable/active document=]'s [=relevant global object=].
+        1. Set |navigable|'s [=top-level traversable/bounce tracking record=] to a new [=bounce tracking record=] with
+            [=bounce tracking record/initial host=] set to |initialHost|.
+    1. Otherwise, add |initialHost| to |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/bounce set=].
+1. Set |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/client bounce detection running=] to false.
+
+</div>
+
+<h5 id="bounce-tracking-mitigations-response-received-monkey-patch">Response Received Monkey
+Patch</h5>
+
+When the response is received at the end of a navigation, fill the [=bounce tracking record/bounce set=].
+
+Insert the following steps in the <a spec="html">navigate</a> algorithm, under "Create navigation params by fetching", after Step 19.7,
+"Wait until either response is non-null...", which is the point that [=response/url list=] becomes available.
+
+1. Run [=process response received for bounce tracking=] given
+    <var ignore>navigable</var> and <var ignore>response</var>'s <var ignore>URL list</var>.
+
+<div algorithm>
+
+To <dfn>process response received for bounce tracking</dfn> given a [=navigable=]
+|navigable| and a [=list=] of [=URLs=] |URLs|, perform the following steps:
+
+1. If |navigable| is not a [=top-level traversable=], then abort these steps.
+1. [=Assert=]: |navigable|'s [=top-level traversable/bounce tracking record=] is not null.
+1. [=Run steps after a timeout=] given:
+    <dl>
+        <dt><var ignore>global</var></dt>
+        <dd>|navigable|'s [=navigable/active document=]'s [=relevant global object=]</dd>
+        <dt><var ignore>milliseconds</var></dt>
+        <dd>[=client bounce detection timer period=]</dd>
+        <dt><var ignore>completionSteps</var></dt>
+        <dd>queue a [=task=] to [=record stateful bounces for bounce tracking=] given <var ignore>global</var></dd>
+    </dl>
+1. Set |navigable|'s [=top-level traversable/bounce tracking record=]'s
+    [=bounce tracking record/client bounce detection running=] to true.
+1. [=list/For each=] |URL| in |URLs|:
+    1. Let |site| be the result of running [=obtain a site=] given |URL|.
+    1. Let |host| be the |site|'s [=host=].
+    1. [=Insert=] |host| to the |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/bounce set=].
+
+</div>
+
+<h5 id="bounce-tracking-mitigations-document-load-monkey-patch">Document Loaded Monkey
```

Comment:
I'm curious why this patch and the "Response Received" patch are separate, and it's probably worth adding a Note about why. It seems like you need the same URL list for each, and that it'd be fine to update the intermediate bounces and start the timer at the same time as you update the "final" host.